app/services/plans/update_service.rb

```diff
@@ -2,63 +2,38 @@
 
 module Plans
   class UpdateService < BaseService
-    def update(**args)
-      plan = result.user.plans.find_by(id: args[:id])
-      return result.not_found_failure!(resource: 'plan') unless plan
-
-      plan.name = args[:name]
-      plan.description = args[:description]
-
-      # NOTE: Only name and description are editable if plan
-      #       is attached to subscriptions
-      unless plan.attached_to_subscriptions?
-        plan.code = args[:code]
-        plan.interval = args[:interval].to_sym
-        plan.pay_in_advance = args[:pay_in_advance]
-        plan.amount_cents = args[:amount_cents]
-        plan.amount_currency = args[:amount_currency]
-        plan.trial_period = args[:trial_period]
-        plan.bill_charges_monthly = args[:interval].to_sym == :yearly ? args[:bill_charges_monthly] || false : nil
-      end
-
-      metric_ids = args[:charges].map { |c| c[:billable_metric_id] }.uniq
-      if metric_ids.present? && plan.organization.billable_metrics.where(id: metric_ids).count != metric_ids.count
-        return result.not_found_failure!(resource: 'billable_metrics')
-      end
-
-      ActiveRecord::Base.transaction do
-        plan.save!
-
-        process_charges(plan, args[:charges])
-      end
+    def self.call(...)
+      new(...).call
+    end
 
-      result.plan = plan
-      result
-    rescue ActiveRecord::RecordInvalid => e
-      result.record_validation_failure!(record: e.record)
+    def initialize(plan:, params:)
+      @plan = plan
+      @params = params
+      super
     end
 
-    def update_from_api(organization:, code:, params:)
-      plan = organization.plans.find_by(code: code)
+    def call
       return result.not_found_failure!(resource: 'plan') unless plan
 
       plan.name = params[:name] if params.key?(:name)
       plan.description = params[:description] if params.key?(:description)
 
+      # NOTE: Only name and description are editable if plan
+      #       is attached to subscriptions
       unless plan.attached_to_subscriptions?
         plan.code = params[:code] if params.key?(:code)
         plan.interval = params[:interval].to_sym if params.key?(:interval)
         plan.pay_in_advance = params[:pay_in_advance] if params.key?(:pay_in_advance)
         plan.amount_cents = params[:amount_cents] if params.key?(:amount_cents)
         plan.amount_currency = params[:amount_currency] if params.key?(:amount_currency)
         plan.trial_period = params[:trial_period] if params.key?(:trial_period)
-        plan.bill_charges_monthly = params[:interval]&.to_sym == :yearly ? params[:bill_charges_monthly] || false : nil
+        plan.bill_charges_monthly = (params[:interval]&.to_sym == :yearly ? params[:bill_charges_monthly] || false : nil)
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)